Feature: Add carousel to product view (Issue #11)

fishtriangle opened the pull request:
Product.ts - add array to photoUrl for multiple images 
Product.tsx - add carousel component
package.json and yarn.lock - add nuka-carousel library

fishtriangle committed:
Add carousel to product view

Product.ts - add array to photoUrl for multiple images
Product.tsx - add carousel component
package.json and yarn.lock - add nuka-carousel library

Loskir commented:
```
./components/ProductListItem.tsx:67:13
09:05:56.918	Type error: Type 'string | string[]' is not assignable to type 'string | undefined'.
```

Build failed unfortunately

Loskir commented on components/ProductView/Product.tsx:
```diff
@@ -14,14 +17,33 @@ import { ProductHistoryGraph } from "./ProductHistoryGraph"
+      renderCenterLeftControls={({ previousSlide }) => <ArrowBackIosIcon onClick={previousSlide} />}
+      renderCenterRightControls={({ nextSlide }) => <ArrowForwardIosIcon onClick={nextSlide} />}
+    >
+      {urls.map((url) => <ProductImage url={url} key={url} cls={"h-[31vh] pb-[35px]"} />)}
```
Why is cls needed?

I think there should be a way to set the height of the carousel itself. And what about paddings? And also why `cls` instead of `className` like in react itself?

Loskir commented on components/ProductView/Product.tsx:
```diff
@@ -14,14 +17,33 @@ import { ProductHistoryGraph } from "./ProductHistoryGraph"
+    <Carousel
+      wrapAround
+      autoplay
+      autoplayInterval={5000}
```
I don't think it should be enabled

fishtriangle committed:
Change carousel height setting

Product.tsx: reverse changes in ProductImage component, disable autoplay of carousel, add className and add wrapAround prop, change children images of carousel
Product.module.css - add carousel classes (flex to align images)
[id].tsx - fix the bug with array in product image srcs

Loskir committed:
Photo url fixes

Loskir committed:
Update carousel size

Loskir committed:
Update product structure

Loskir committed:
Update padding

Loskir commented:
I think it's now ready, thank you so much for your help!
